### PR TITLE
PyTorch API updates

### PR DESCRIPTION
**Pull request by bcebere**
- fix train/fit confusion
 - bump version to 0.2.1 since the API is changed.
 - GPU fixes
 - enable more tests

fixes https://github.com/AliciaCurth/CATENets/issues/19

### CODE REVIEW

**Comment by AliciaCurth** [comment]
Just to check, does this actually get rid of all API inconsistencies? (Looking at @kamilest's original issue, she mentioned that torch functions need to be  in 'train-mode' --> is this now the case during the 'fit' call? I.e. is train mode the default behaviour? If not, that probably needs to be fixed? If on the other hand train mode _is_ the default, then the predict functions probably all have to be set into eval mode to remove dropout etc, correct?)

File: catenets/models/torch/pseudo_outcome_nets.py [hunk 1]
```diff
@@ -331,8 +329,8 @@ def predict(self, X: torch.Tensor, return_po: bool = False, training: bool = Fal
             raise NotImplementedError(
                 "PseudoOutcomeLearners have no Potential outcome predictors."
             )
-        if not training:
-            self._te_estimator.model.eval()
+        if not training and hasattr(self._te_estimator, "eval"):
```
Comment (AliciaCurth) [inline catenets/models/torch/pseudo_outcome_nets.py:332]: Why do we have the setting to Eval mode only for SLearner? I think we'd probably need it for all methods? (@kamilest can you comment on this?)

**Comment by bcebere** [comment]
> Just to check, does this actually get rid of all API inconsistencies? (Looking at @kamilest's original issue, she mentioned that torch functions need to be in 'train-mode' --> is this now the case during the 'fit' call? I.e. is train mode the default behaviour? If not, that probably needs to be fixed? If on the other hand train mode _is_ the default, then the predict functions probably all have to be set into eval mode to remove dropout etc, correct?)

The original issues was that the APIs were shadowing the `train` method inherited from the nn.Module. By default, self.model is in train mode, you need to call train() only if you re-train it. But we can explicitly call `.train()` on every fit() anyway.
